File: frontend/src/api/ai.ts
import apiClient from './index'

export interface AssertionSuggestion {
  type: 'status' | 'jsonpath' | 'header' | 'regex'
  field: string
  expected: any
  description?: string
  confidence: number
}

export interface TestCaseSuggestion {
  name: string
  description?: string
  method: string
  url: string
  headers?: Record<string, string>
  body?: any
  assertions: AssertionSuggestion[]
  test_data?: Record<string, any>
  reasoning?: string
}

export interface AnomalyAnalysis {
  error_type: string
  severity: 'critical' | 'high' | 'medium' | 'low'
  root_cause: string
  suggestion: string
  related_tests?: number[]
  confidence: number
}

export interface ApiInfo {
  path: string
  method: string
  summary?: string
  description?: string
  parameters?: any[]
  request_body?: any
  responses?: any
}

export interface AIStatus {
  provider: string
  available: boolean
  model?: string
  error?: string
}

export const aiApi = {
  // 获取 AI 服务状态
  getStatus: (): Promise<AIStatus> => {
    return apiClient.get('/ai/status')
  },

  // AI 生成测试用例
  generateTestCases: (apiInfo: ApiInfo, count: number = 3): Promise<TestCaseSuggestion[]> => {
    return apiClient.post('/ai/generate-test-cases', {
      api_info: apiInfo,
      count
    })
  },

  // AI 生成断言
  generateAssertions: (
    apiResponse: Record<string, any>,
    context?: Record<string, any>
  ): Promise<AssertionSuggestion[]> => {
    return apiClient.post('/ai/generate-assertions', {
      api_response: apiResponse,
      context
    })
  },

  // AI 分析异常
  analyzeAnomaly: (
    testResult: Record<string, any>,
    relatedContext?: Record<string, any>
  ): Promise<AnomalyAnalysis> => {
    return apiClient.post('/ai/analyze-anomaly', {
      test_result: testResult,
      related_context: relatedContext
    })
  },

  // AI 改进建议
  suggestImprovements: (
    testCaseId: number,
    recentResults: Record<string, any>[]
  ): Promise<string[]> => {
    return apiClient.post('/ai/suggest-improvements', {
      test_case_id: testCaseId,
      recent_results: recentResults
    })
  }
}

File: backend/app/schemas/ai.py
"""
AI-related Pydantic schemas.
"""
from pydantic import BaseModel
from typing import List, Optional, Dict, Any


class ApiInfo(BaseModel):
    """API 信息"""
    path: str
    method: str
    summary: Optional[str] = None
    description: Optional[str] = None
    parameters: Optional[List[Dict[str, Any]]] = None
    request_body: Optional[Dict[str, Any]] = None
    responses: Optional[Dict[str, Any]] = None


class AssertionSuggestion(BaseModel):
    """AI 生成的断言建议"""
    type: str
    field: str
    expected: Any
    description: Optional[str] = None
    confidence: float = 0.8


class TestCaseSuggestion(BaseModel):
    """AI 生成的测试用例建议"""
    name: str
    description: Optional[str] = None
    method: str
    url: str
    headers: Optional[Dict[str, str]] = None
    body: Optional[Any] = None
    assertions: List[AssertionSuggestion] = []
    test_data: Optional[Dict[str, Any]] = None
    reasoning: Optional[str] = None


class AnomalyAnalysis(BaseModel):
    """AI 异常分析结果"""
    error_type: str
    severity: str
    root_cause: str
    suggestion: str
    related_tests: Optional[List[int]] = None
    confidence: float = 0.8


class GenerateTestCasesRequest(BaseModel):
    """生成测试用例请求"""
    api_info: ApiInfo
    count: int = 3


class GenerateAssertionsRequest(BaseModel):
    """生成断言请求"""
    api_response: Dict[str, Any]
    context: Optional[Dict[str, Any]] = None


class AnalyzeAnomalyRequest(BaseModel):
    """分析异常请求"""
    test_result: Dict[str, Any]
    related_context: Optional[Dict[str, Any]] = None


class SuggestImprovementsRequest(BaseModel):
    """建议改进请求"""
    test_case_id: int
    recent_results: List[Dict[str, Any]]


class AIProviderConfig(BaseModel):
    """AI Provider 配置"""
    provider: str = "openai"
    api_key: Optional[str] = None
    model: Optional[str] = None
    base_url: Optional[str] = None


class AIStatus(BaseModel):
    """AI 服务状态"""
    provider: str
    available: bool
    model: Optional[str] = None
    error: Optional[str] = None
